tests/test_dataset_manager.py

```
@@ -88,6 +104,39 @@ def drop_duplicates(df):
     clean_up_test(dsm, rfp) #leave commented out until we figure out reset
     #dsm.post_dataset("my_test")
 
+def test_no_header(bad_config_manager_header):
+    """
+    Test that DSM invalidates dataset with no header.
+    """
+    error_message = ("No header has been provided in file {file} in folder "
+                     "{folder}")
+    try:
+        dsm = DatasetManager(bad_config_manager_header)
+        assert False, "Error should have been thrown for lack of header"
```

Can you raise an exception here too?

---

Sure.